client/state/reader/streams/reducer.js

```diff
@@ -0,0 +1,37 @@
+/**
```

Comment:
FFTI, but I think this file could make really good use of both:
1. `keyedReducer`
2. `createReducer`